src/txgraph.cpp

```diff
@@ -1675,6 +1677,8 @@ void Cluster::GetClusterRefs(TxGraphImpl& graph, std::span<TxGraph::Ref*> range,
         Assume(entry.m_ref != nullptr);
         ref = entry.m_ref;
     }
+    // Return whether this was the end of the Cluster.
+    return start_pos == m_linearization.size();
```

Comment:
nit: fwiw forgot start_pos was actually end_pos by this point, maybe this function should have cur_pos?

Reply:
feel free to resolve